src/state/opamEnv.ml

```diff
@@ -378,9 +378,13 @@ let shell_eval_invocation shell cmd =
 
 (** Returns "opam env" invocation string together with optional root and switch
     overrides *)
-let opam_env_invocation ?root ?switch ?(set_opamswitch=false) () =
-  let root = OpamStd.Option.map_default (Printf.sprintf " --root=%s") "" root in
-  let switch = OpamStd.Option.map_default (Printf.sprintf " --switch=%s") "" switch in
+let opam_env_invocation ?root ?switch ?(set_opamswitch=false) shell =
+  let quoted_arg arg = match shell with
+    | SH_win_powershell | SH_pwsh -> Printf.sprintf " --%s=\"%s\"" arg
+    | _ -> Printf.sprintf " --%s='%s'" arg
```

Comment:
Could you list them explicitly to avoid feature issues?
```suggestion
    | SH_win_powershell | SH_pwsh -> Printf.sprintf " --%s=\"%s\"" arg
    | All | The | Others -> Printf.sprintf " --%s='%s'" arg
```

Reply:
Yes. Will do.